src/coreclr/vm/stackwalk.cpp

```diff
@@ -2882,6 +2877,15 @@ void StackFrameIterator::ProcessCurrentFrame(void)
                     SyncRegDisplayToCurrentContext(pRD);
                 }
             }
+            else if (InlinedCallFrame::FrameHasActiveCall(m_crawl.pFrame) && m_crawl.pFrame->PtrNextFrame() != FRAME_TOP && m_crawl.pFrame->PtrNextFrame()->GetFrameIdentifier() == FrameIdentifier::InterpreterFrame)
+            {
+                // There is an active inlined call frame and the next frame is the interpreter frame. This is a special case where we need to save the current context registers that the interpreter frames walking reuses.
+                m_interpExecMethodIP = (TADDR)GetIP(pRD->pCurrentContext);
+                m_interpExecMethodSP = (TADDR)GetSP(pRD->pCurrentContext);
+                m_interpExecMethodFP = (TADDR)GetFP(pRD->pCurrentContext);
+                m_interpExecMethodFirstArgReg = (TADDR)GetFirstArgReg(pRD->pCurrentContext);
```

Comment:
Does this need a cast? Doesn't GetFirstArgReg return TADDR?

Reply:
You are right, the casts are not needed. I don't know why I have added them. I'll remove them.